### PR TITLE
Add support for Activity Summary

### LINKED ISSUE #118: Support querying activity goals (move, exercise, stand)
**Is your feature request related to a problem? Please describe.**
Accessing these goals is an essential part of our app

**Describe the solution you'd like**
We can call a function or functions to retrieve these values

**Describe alternatives you've considered**
Have our users manually set their goals separately in our app

### PR DESCRIPTION
**Pull request by KyleRosenberg**
## Description

Adds support for [HKActivitySummary](https://developer.apple.com/documentation/healthkit/hkactivitysummary), specifically the values in the 3 rings (move, exercise, stand, and their goals).

Fixes #118 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: RCTAppleHealthKit/RCTAppleHealthKit+Methods_Summary.m [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{
+    NSDate *startDate = [RCTAppleHealthKit dateFromOptions:input key:@"startDate" withDefault:nil];
+    NSDate *endDate = [RCTAppleHealthKit dateFromOptions:input key:@"endDate" withDefault:[NSDate date]];
+
+    void (^completion)(NSArray *results, NSError *error);
+
+    completion = ^(NSArray *results, NSError *error) {
+        if (results){
+            
+            callback(@[[NSNull null], results]);
```
Comment (macelai) [inline RCTAppleHealthKit/RCTAppleHealthKit+Methods_Summary.m:26]: Can we get rid of these blank lines, so it's the same style as others?

File: src/constants/Permissions.js [hunk 2]
```diff
@@ -77,4 +77,5 @@ export const Permissions = {
   WaistCircumference: 'WaistCircumference',
   Weight: 'Weight',
   Workout: 'Workout',
+  Summary: "Summary"
```
Comment (macelai) [inline src/constants/Permissions.js:80]: What do you think of calling this "Activity Summary"?

**Review by macelai — CHANGES_REQUESTED** [review]

**Comment by macelai** [comment]
Thank you for this amazing job!

**Comment by macelai** [comment]
Can you also add this doc to the `permissions.md`?

File: RCTAppleHealthKit/RCTAppleHealthKit+Methods_Summary.m (hunk 1 above)
Comment (KyleRosenberg) [reply RCTAppleHealthKit/RCTAppleHealthKit+Methods_Summary.m:25]: Fixed

File: src/constants/Permissions.js (hunk 2 above)
Comment (KyleRosenberg) [reply src/constants/Permissions.js]: Sounds good, fixed

**Review by macelai — APPROVED** [review]

**Comment by macelai** [comment]
Also?

**Comment by anhkieet** [comment]
Hi @KyleRosenberg Can you brief me on how to use this function?

**Comment by KyleRosenberg** [comment]
@anhkieet Have you read the docs? https://github.com/agencyenterprise/react-native-health/blob/master/docs/getActivitySummary.md

**Comment by ovaris** [comment]
@KyleRosenberg so how do I know for which day each activity summary object in resulting array is?

**Comment by KyleRosenberg** [comment]
@ovaris They are in order from start to end

**Comment by krakz999** [comment]
Is there a way to add Exercise Time to Apple Health in a simulator?

**Comment by justinkgibbons89** [comment]
> @ovaris They are in order from start to end

@KyleRosenberg What if there are days without data? How do you know if N-5 is 5 days ago or 20 days ago?

**Pull request merged**